models/utils/cohort_template_id_format.py

```diff
@@ -0,0 +1,81 @@
+from typing import Iterable
+
+from api.settings import COHORT_TEMPLATE_CHECKSUM_OFFSET, COHORT_TEMPLATE_PREFIX
+from models.utils.luhn import luhn_compute, luhn_is_valid
+
+
+def cohort_template_id_format(cohort_template_id: int | str) -> str:
+    """
+    Transform raw (int) cohort template identifier to format (CTPLXXX) where:
+        - CTPL is the prefix
+        - XXX is the original identifier
+        - H is the Luhn checksum
+    """
+    # Validate input
+    if isinstance(cohort_template_id, str) and not cohort_template_id.isdigit():
+        if cohort_template_id.startswith(COHORT_TEMPLATE_PREFIX):
+            return cohort_template_id
+        raise ValueError(f'Unexpected format for cohort template identifier {cohort_template_id!r}')
+
+    cohort_template_id = int(cohort_template_id)
+
+    checksum = luhn_compute(
+        cohort_template_id, offset=COHORT_TEMPLATE_CHECKSUM_OFFSET
+    )
+
+    return f'{COHORT_TEMPLATE_PREFIX}{cohort_template_id}{checksum}'
+
+
+def cohort_template_id_format_list(cohort_template_ids: Iterable[int | str]) -> list[str]:
+    """
+    Transform LIST of raw (int) cohort template identifier to format (CTPLXXX) where:
+        - CTPL is the prefix
+        - XXX is the original identifier
+        - H is the Luhn checksum
+    """
+    return [cohort_template_id_format(s) for s in cohort_template_ids]
+
+
+def cohort_template_id_transform_to_raw(cohort_template_id: int | str) -> int:
```

Comment:
YP: Can we reuse function 'cohort_id_transform_to_raw' as the implementation is the same, the only diff is PREFIX, which can be passed as an argument.

Reply:
Agree that we can refactor to remove a lot of redundant code between all the format_id scripts. 
I've created a relevant issue and I think it's better to handle this outside of this PR :) 
https://github.com/populationgenomics/metamist/issues/736